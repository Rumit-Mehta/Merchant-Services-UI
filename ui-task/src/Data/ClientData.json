[
  {
    "clientId": "000001",
    "clientName": "Corporation Name"
  }
]
