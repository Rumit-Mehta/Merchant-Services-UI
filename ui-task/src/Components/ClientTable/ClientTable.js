import React from "react";
import { withStyles, makeStyles } from "@material-ui/core/styles";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";
import ClientData from "../../Data/ClientData.json";

const StyledTableCell = withStyles((theme) => ({
  head: {
    color: theme.palette.primary.main,
    fontWeight: "bold",
  },
  body: {
    color: theme.palette.common.black,
  },
}))(TableCell);

export { StyledTableCell };

const useStyles = makeStyles({
  table: {
    minWidth: 650,
  },
});

export default function SimpleTable() {
  const classes = useStyles();

  return (
    <Paper elevation={0}>
      <TableContainer>
        <Table className={classes.table} aria-label="simple table">
          <TableBody>
            {ClientData.map((row) => (
              <TableRow key={row.cleintId}>
                <StyledTableCell component="th" scope="row">
                  <strong>Client Name</strong>
                </StyledTableCell>
                <StyledTableCell align="left">{row.clientName}</StyledTableCell>
              </TableRow>
            ))}
            {ClientData.map((row) => (
              <TableRow key={row.cleintId}>
                <StyledTableCell component="th" scope="row">
                  <strong>Client Id</strong>
                </StyledTableCell>
                <StyledTableCell align="left">{row.clientId}</StyledTableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </TableContainer>
    </Paper>
  );
}
