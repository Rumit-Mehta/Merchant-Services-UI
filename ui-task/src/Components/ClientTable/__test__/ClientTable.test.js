import React from "react";
import ReactDOM from "react-dom";
import ClientTable from "../ClientTable";
import { cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import { TableRow } from "@material-ui/core";
import ClientData from "../../../Data/ClientData.json";
import { StyledTableCell } from "../ClientTable";

afterEach(cleanup);
test("renders  without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<ClientTable />, div);
});

test("table has correct rows and columns", () => {
  const wrapper = shallow(<ClientTable />);
  const rows = wrapper.find(TableRow);
  expect(rows).toHaveLength(2);

  const dataEntries = wrapper.find(StyledTableCell);
  const columns = 2;
  expect(dataEntries.length / rows.length).toEqual(columns);
});

test("matches snapshot", () => {
  const tree = renderer.create(<ClientTable />).toJSON();
  expect(tree).toMatchSnapshot();
});
