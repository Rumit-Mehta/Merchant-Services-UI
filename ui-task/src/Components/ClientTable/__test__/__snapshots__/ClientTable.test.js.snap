// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation0 MuiPaper-rounded"
>
  <div
    className="MuiTableContainer-root"
  >
    <table
      aria-label="simple table"
      className="MuiTable-root makeStyles-table-1"
      role={null}
    >
      <tbody
        className="MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3"
            role="cell"
            scope="row"
          >
            <strong>
              Client Name
            </strong>
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignLeft"
          >
            Corporation Name
          </td>
        </tr>
        <tr
          className="MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3"
            role="cell"
            scope="row"
          >
            <strong>
              Client Id
            </strong>
          </th>
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body WithStyles(ForwardRef(TableCell))-body-3 MuiTableCell-alignLeft"
          >
            000001
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
